src/stores/right-panel/RightPanelStore.ts

```diff
@@ -0,0 +1,334 @@
+        const roomCache = this.byRoom[rId];
+        roomCache.history.pop();
+        this.emitAndUpdateSettings();
+    }
+
+    public togglePanel(roomId: string = null) {
+        const rId = roomId ?? this.viewedRoomId;
+        this.byRoom[rId].isOpen = !this.byRoom[rId].isOpen;
+        this.emitAndUpdateSettings();
+    }
+
+    // Private
+
+    private loadCacheFromSettings() {
+        const room = this.mxClient?.getRoom(this.viewedRoomId);
+        if (!!room) {
+            this.global = this.global ?? convertToStateRoom(SettingsStore.getValue("RightPanel.phasesGlobal"), room);
+            this.byRoom[this.viewedRoomId] = this.byRoom[this.viewedRoomId]
+            ?? convertToStateRoom(SettingsStore.getValue("RightPanel.phases", this.viewedRoomId), room);
+        }
+    }
+    private emitAndUpdateSettings() {
+        const storePanelGlobal = convertToStoreRoom(this.global);
+        const storePanelThisRoom = convertToStoreRoom(this.byRoom[this.viewedRoomId]);
+        SettingsStore.setValue("RightPanel.phasesGlobal", null, SettingLevel.DEVICE, storePanelGlobal);
+        if (!!this.viewedRoomId) {
+            SettingsStore.setValue("RightPanel.phases",
+                this.viewedRoomId,
+                SettingLevel.ROOM_DEVICE,
+                storePanelThisRoom,
+            );
+        }
+        this.emit(UPDATE_EVENT, null);
+    }
+
+    // set rigth panel: overrides the history
+    private setRightPanelCache(targetPhase: RightPanelPhases, panelState: IPanelState = null) {
+        this.byRoom[this.viewedRoomId] = {
+            history: [{ phase: targetPhase, state: panelState ?? {} }],
+            isOpen: true,
+        };
+        this.emitAndUpdateSettings();
+    }
+
+    // CHECKS:
+    //  - RoomMemberInfo -> needs to be changed to EncryptionPanel if pendingVerificationRequestForUser
+    private getVerificationRedirect(targetPhase: RightPanelPhases, panelState):
+        { targetPhase: RightPanelPhases, panelState } {
+        if (targetPhase === RightPanelPhases.RoomMemberInfo && panelState) {
+            const { member } = panelState;
+            const pendingRequest = pendingVerificationRequestForUser(member);
+            if (pendingRequest) {
+                return {
+                    targetPhase: RightPanelPhases.EncryptionPanel,
+                    panelState: {
+                        verificationRequest: pendingRequest,
+                        member,
+                    },
+                };
+            }
+        }
+        return null;
+    }
+
+    private isPhaseActionIsValid(targetPhase) {
+        if (!RightPanelPhases[targetPhase]) {
+            logger.warn(`Tried to switch right panel to unknown phase: ${targetPhase}`);
+            return false;
+        }
+        if (GROUP_PHASES.includes(targetPhase) && this.isViewingRoom) {
+            logger.warn(`Tried to switch right panel to a group phase: ${targetPhase}, 
+            but we are currently not viewing a group`);
```

Comment:
Multi-line strings need to be per line and concatenated, this style includes all the indentation in the string.

```suggestion
            logger.warn(
                `Tried to switch right panel to a group phase: ${targetPhase}, ` +
                `but we are currently not viewing a group`,
            );
```